content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-camera/giga-camera.md

```diff
@@ -110,188 +135,137 @@ This sketch is also available in the Arduino IDE via **Examples > Camera > Camer
+    byte* buffer = fb.getBuffer();
+    size_t bufferSize = cam.frameSize();
+    digitalWrite(LED_BUILTIN, LOW);
+    
+    sendChunk(START_SEQUENCE, sizeof(START_SEQUENCE));
+
+    // Split buffer into chunks
+    for(size_t i = 0; i < bufferSize; i += CHUNK_SIZE) {
+      size_t chunkSize = min(bufferSize - i, CHUNK_SIZE);
+      sendChunk(buffer + i, chunkSize);
+    }    
+    
+    sendChunk(STOP_SEQUENCE, sizeof(STOP_SEQUENCE));
+    
+    digitalWrite(LED_BUILTIN, HIGH);
   } else {
     blinkLED(20);
   }
 }
 
-```
-
-### Step 2: Processing
-
-The following Processing sketch will launch a Java app that allows you to view the camera feed. As data is streamed via serial, make sure you close the Serial Monitor during this process, else it will not work.
-
-***Important! Make sure to replace the following line in the code below: `/dev/cu.usbmodem14301`, with the name of your port.***
-
-Click on the **"PLAY"** button to initialize the app.
-
-```cpp
-/*
-  Use with the Examples -> CameraCaptureRawBytes Arduino sketch.
-  This example code is in the public domain.
-*/
-
-import processing.serial.*;
-import java.nio.ByteBuffer;
-import java.nio.ByteOrder;
-
-Serial myPort;
+/**
+ * Sends the camera configuration over a serial connection.
+ * This is used to configure the web app to display the image correctly.
+ */
+void sendCameraConfig(){
+  Serial.write(IMAGE_MODE);
+  Serial.write(RESOLUTION);
+  Serial.flush();
+  delay(1);
+}
 
-// must match resolution used in the Arduino sketch
-final int cameraWidth = 320;
-final int cameraHeight = 240;
+void loop() {
+  if(!Serial) {    
+    Serial.begin(115200);
+    while(!Serial);    
+  }
 
-// Must match the image mode in the Arduino sketch
-final boolean useGrayScale = true;
+  if(!Serial.available()) return;
 
-// Must match the baud rate in the Arduino sketch
-final int baudRate = 921600;
+  byte request = Serial.read();
 
-final int cameraBytesPerPixel = useGrayScale ? 1 : 2;
-final int cameraPixelCount = cameraWidth * cameraHeight;
-final int bytesPerFrame = cameraPixelCount * cameraBytesPerPixel;
-final int timeout =  int((bytesPerFrame / float(baudRate / 10)) * 1000 * 2); // Twice the transfer rate
+  switch(request){
+    case IMAGE_SEND_REQUEST:
+      sendFrame();
+      break; 
+    case CONFIG_SEND_REQUEST:
+      sendCameraConfig();
+      break;
+  }
+  
+}
 
-PImage myImage;
-byte[] frameBuffer = new byte[bytesPerFrame];
-int lastUpdate = 0;
-boolean shouldRedraw = false;
+```
 
-void setup() {
-  size(640, 480);  
+### Step 2: Web Serial
 
-  // If you have only ONE serial port active you may use this:
-  //myPort = new Serial(this, Serial.list()[0], baudRate);          // if you have only ONE serial port active
+Open the [Webserial Interface](https://labs.oniudra.cc/en/labs/web-serial-camera) which allows you to view the camera feed. As data is streamed via serial, make sure you close the Serial Monitor during this process, else it will not work.
```

Comment:
```suggestion
Open the [Web Serial Interface](https://arduino.github.io/labs-pages/web-serial-camera/) which allows you to view the camera feed. Make sure to close the Serial Monitor in the Arduino IDE beforehand, otherwise it will not work as the serial port would be occupied.
```